ICP.py

```diff
@@ -0,0 +1,109 @@
+import torch
+import pypose as pp
+from torch import nn
+
+
+class ICP(nn.Module):
+    r'''
+    Iterative Closest Point (ICP) using Singular Value Decomposition (SVD).
+
+    Args:
+        max_iterations: the max iteration number
+        tolerance: the tolerance of error for early stopping
+        is_matching: whether the input points set and the target points set have been matched
+    '''
+
+    def __init__(self, max_iterations=200, tolerance=0.0001, is_matching=False):
+        super().__init__()
+        self.max_iterations = max_iterations
+        self.tolerance = tolerance
+        self.is_matching = is_matching
+
+    def forward(self, p1, p2):
+        ori_p2 = p2.clone()
+        temp_pc = p1.clone()
+        iter_num = 0
+        prev_err = 0
+        if (not self.is_matching):
+            while iter_num <= self.max_iterations:
+                iter_num += 1
+                nn = self.nearest_neighbor(temp_pc, p2)
+                mean_err = nn.values.mean()
+                transR, transT = self.get_transform(temp_pc, ori_p2[:, nn.indices[-1], :].squeeze(-2))
+                temp_pc = temp_pc @ transR + transT
+                if (abs(prev_err - mean_err) < self.tolerance):
+                    break
+                prev_err = mean_err
+
+            transR, transT = self.get_transform(p1, temp_pc)
+            transT = torch.transpose(transT, 1, 2)
+            T = torch.cat([transR, transT], dim=2)
+            return pp.mat2SE3(T, check=False)
+        else:
+            transR, transT = self.get_transform(p1, p2)
+            transT = torch.transpose(transT, 1, 2)
+            T = torch.cat([transR, transT], dim=2)
+            return pp.mat2SE3(T)
+
+    def get_transform(self, p1: torch.tensor, p2: torch.tensor):
```

Comment:
There is an existing transform calculation module just merged to main branch. Consider using it so that we don't have to keep two copies of the same functionality
https://github.com/pypose/pypose/blob/60a38038cc6b18811749084b85808bde510357fb/pypose/module/pnp.py#L294